### ISSUE TITLE
关于模型输出的疑问

### ISSUE THREAD

**Issue by Xixi-x-i** [sgoerttler/MSA-CNN#11]
你好，请问为什么模型训练阶段，训练每一个epoch都会输出测试集的性能，不应该是100个epoch训练结束，再评估测试集吗

**Comment by sgoerttler** [comment]
Hello, this information is saved for plotting the training and test accuracy against the number of epochs. It would be better to use an external validation dataset for hyperparameter validation, but the nature of the experiment (deep learning in combination with cross validation with hierarchical data) means that this is computationally expensive. It is therefore a good idea to report a range of hyperparameter settings.

您好，保存此信息是为了绘制训练和测试准确率与周期数的关系图。最好使用外部验证数据集进行超参数验证，但实验的性质（深度学习与分层数据的交叉验证相结合）意味着这在计算上很昂贵。因此，报告一系列超参数设置是个好主意。

**Comment by Xixi-x-i** [comment]
请问您论文中的结果是怎么计算的呢，我复现您的代码，只得到10折叠每一个epoch输出的训练和测试的准确率和损失值，请问您怎么计算它们的总体性能呢?

**Comment by sgoerttler** [comment]
I calculate the weighted average across all folds, which yields the results (training and test accuracy) for one run. Then I average across 10 runs. The overall performance is the mean test accuracy across the 10 runs. If I have more time I will upload the analysis as well. I ran the models on different computers and with older versions of the code, so the analysis file is very messy.

我计算所有折叠的加权平均值，得出一次运行的结果（训练和测试准确率）。然后我计算 10 次运行的平均值。总体表现是 10 次运行的平均测试准确率。如果我有更多时间，我也会上传分析结果。我在不同的计算机上使用旧版本的代码运行了模型，因此分析文件非常混乱。

**Comment by Xixi-x-i** [comment]
请问您计算所有折叠的加权平均值时，是使用的每折叠最后一个epoch测试的结果吗，还是使用每个折叠100个epoch中最好的测试结果

**Comment by sgoerttler** [comment]
I use the value for the last epoch only. To select the best value out of the 100 epochs, one needs an additional validation set that is distinct from the testing set. This would require the use of nested cross validation or additional data.

我只使用最后一个 epoch 的值。为了从 100 个 epoch 中选出最佳值，需要一个与测试集不同的额外验证集。这需要使用嵌套交叉验证或额外的数据。
